packages/contracts-bedrock/src/L1/DelayedVetoable.sol

```diff
@@ -0,0 +1,159 @@
+    // @notice The address that can initiate a call.
+    address internal _initiator;
+
+    /// @notice The time that a call was initiated.
+    mapping(bytes32 => uint256) internal _queuedAt;
+
+    // @notice The time to wait before forwarding a call.
+    uint256 internal _delay;
+
+    /// @notice A modifier that reverts if not called by the vetoer or by address(0) to allow
+    ///         eth_call to interact with this proxy without needing to use low-level storage
+    ///         inspection. We assume that nobody is able to trigger calls from address(0) during
+    ///         normal EVM execution.
+    modifier handleCallIfNotVetoer() {
+        if (msg.sender == _vetoer || msg.sender == address(0)) {
+            _;
+        } else {
+            // This WILL halt the call frame on completion.
+            _handleCall();
+        }
+    }
+
+    /// @custom:semver 0.0.1
+    /// @notice Sets the target admin during contract deployment.
+    /// @param vetoer_ Address of the vetoer.
+    /// @param initiator_ Address of the initiator.
+    /// @param target_ Address of the target.
+    /// @param delay_ Address of the delay.
+    constructor(address vetoer_, address initiator_, address target_, uint256 delay_) Semver(0, 1, 1) {
+        _vetoer = vetoer_;
+        _initiator = initiator_;
+        _target = target_;
+        _delay = delay_;
+    }
+
+    /// @notice Gets the initiator
+    /// @return Initiator address.
+    function initiator() external handleCallIfNotVetoer returns (address) {
+        return _initiator;
+    }
+
+    //// @notice Queries the vetoer address.
+    /// @return Vetoer address.
+    function vetoer() external handleCallIfNotVetoer returns (address) {
+        return _vetoer;
+    }
+
+    //// @notice Queries the target address.
+    /// @return Target address.
+    function target() external handleCallIfNotVetoer returns (address) {
+        return _target;
+    }
+
+    /// @notice Gets the delay
+    /// @return Delay address.
+    function delay() external handleCallIfNotVetoer returns (uint256) {
+        return _delay;
+    }
+
+    // TODO(maurelian): Remove this? The contract currently cannot handle forwarding ETH and I'm
+    //   not sure the complexity is warranted.
+    //   If we do allow it:
+    //      1. the callHash will need to include the value
+    //      2. forwarding will need to be done by passing the callHash, rather than the unhashed data
+    /// @notice Used when no data is passed to the contract.
+    receive() external payable {
+        _handleCall();
+    }
+
+    /// @notice Used for all calls that pass data to the contract.
+    fallback() external payable {
+        _handleCall();
+    }
+
+    /// @notice Vetoes a call. This method can only be called by the vetoer. If called by another
+    ///         address, execution will be redirected to _handleCall()
+    function veto(bytes memory data) external handleCallIfNotVetoer {
```

Comment:
Do you have strong preference for this being in the abi? I suppose it does simplify things when interacting with the contract but I imagined it being completely transparent

Reply:
This made the handleCall func a bit cleaner, but I'm not strongly opinionated. 

It's also already not perfectly transparent due to the getters.

Reply:
I took out this method.